### ISSUE TITLE
Placing large numbers of pre-sighted artillery auto-hit hexes causes performance issues

### ISSUE THREAD

**Issue by repligator** [MegaMek/megamek#6041]
### Prerequisites and Pre-Issue Checklist

- [X] I'm reporting the issue to the correct repository:
- [MegaMek](https://github.com/MegaMek/megamek/issues)
- [MegaMekLab](https://github.com/MegaMek/megameklab/issues)
- [MekHQ](https://github.com/MegaMek/mekhq/issues)

- [X] I've tested the issue against at least the latest MILESTONE version
- [X] I've asked on the MegaMek Discord about the error
- [X] I've reviewed the BattleTech rules and MegaMek documentation, and I've confirmed that something isn't working as intended.
- [X] I've searched the Github tracker and haven't found the issue listed

### Severity *

High (Major Disruption): A major feature is broken or incorrect, but a workaround exists.

### Brief Description *

When placing large numbers of pre-sighted artillery auto-hit hexes, performance nose dives in proportion to the number of hexes marked. After the hex designation phase ends, performance returns to normal.

### Steps to Reproduce

1. Load MHQ save. No customs needed.
2. Start the pending convoy mission.
3. Reassign all the offboard artillery to the player (CBS3025)
4. Begin marking hexes.
5. Note the performance issues increasing as the marked hexes increase.

### Operating System *

Linux

### Java Version *

17

### MegaMek Suite Version *

Free Text (type manually)

### Custom MegaMek Version

https://github.com/MegaMek/megamek/commit/72f857dfe36e0b762ea6736dd51112b924e73a11

### Attach Files

[CBS302530250409.cpnx.gz](https://github.com/user-attachments/files/17165174/CBS302530250409.cpnx.gz)
[megamek.log](https://github.com/user-attachments/files/17165185/megamek.log)

### Final Checklist

- [X] I've checked to make sure that this issue has not already been filed
- [X] I'm reporting only one issue in this ticket for clarity and focus

**Comment by repligator** [comment]
Another example, which may work better (and by better I mean worse) for those with more powerful systems. 65 hexes available for designation.
[CBS302530250609.cpnx.gz](https://github.com/user-attachments/files/17167037/CBS302530250609.cpnx.gz)

**Comment by repligator** [comment]
It's hard to describe. After the tenth hex or so, you start to getting a delay between clicking and the action taking place. This delay grows and grows until eventually the whole program freezes. It feels similar to an earlier bug (since fixed) with the missed artillery markers. I will test this under the latest nightly again in a couple minutes.

**Comment by repligator** [comment]
Huh. Well, I just placed 64 auto hit hexes without issues. Two potential explanations.

1. Something else changed in MM since 3 weeks ago that fixed the issue.
2. In the the time between the original report and right now, I upgraded my system from 16GB RAM to 32GB. Perhaps MM misbehaves when it is pushing up against the max system RAM while placing artillery?
3. My old 8GB sticks were going bad and would occasionally cause my whole system to freeze hard and fast. Perhaps MM just had a strange interaction with the bad stick(s)?

Regardless, it is no longer an issue. Closing.
